agent/agent.go

```diff
@@ -164,20 +169,23 @@ type agent struct {
 	// Defaults to statusQueryReplyTimeout if unspecified
 	statusQueryReplyTimeout time.Duration
 
-	// memberIndex is used to select the next member to collect timeline data
-	// from.
-	// Type uint16 so limited to 0 - 65535. Assuming clusters will not have
-	// more than that many members.
-	memberIndex uint16
+	// lastSeen keeps track of the last seen timestamp from a cluster member.
+	// The last seen timestamp can be queried by a member and be used to
+	// filter out events that have already been recorded by this member.
+	lastSeen map[string]time.Time
```

Comment:
Do members ever get removed from this map? Wonder if you considered using TTL map here.